### PR TITLE
chore: add util to generate breakpoint stories

### PR DESCRIPTION
**Pull request by jcfranco**
**Related Issue:** N/A

## Summary

Adds helper to ease adding responsive layout screenshot tests.

### CODE REVIEW

File: packages/calcite-components/.storybook/utils.tsx [hunk 1]
```diff
@@ -139,3 +142,69 @@ export const filterComponentAttributes = (
...
+ *   This should only be used if multiple stories cannot be displayed side-by-side.
+ */
+export function createBreakpointStories(
+  singleStoryHtml: string,
+  focused?: { breakpoint: keyof Breakpoints["width"]; scale: Scale }
+): string {
+  // we hard-code breakpoint values because we can't read them directly from the page when setting up a story
+  // based on https://github.com/Esri/calcite-design-tokens/blob/2e8fc1b8f410b5443fa53ca1c12ceef71e651b9a/tokens/core.json#L1533-L1553
+  const widthBreakpoints: { name: keyof Breakpoints["width"]; maxWidth: number }[] = [
+    { name: "xsmall", maxWidth: 476 },
```
Comment (driskull) [inline packages/calcite-components/.storybook/utils.tsx:160]: Once we can get these values from the tokens we should reference them here.

File: packages/calcite-components/.storybook/utils.tsx [hunk 2]
```diff
@@ -139,3 +142,69 @@ export const filterComponentAttributes = (
...
+            ${singleStoryHtml.replace(placeholderPattern, (_match, placeholder: string) =>
+              placeholder === "scale" ? scale : placeholder
+            )}
+          </div>`;
+        });
+    });
+
+  return html`<div class="${css.storiesContainer}">
+    <style>
+      .${css.storiesContainer} {
```
Comment (driskull) [inline packages/calcite-components/.storybook/utils.tsx:193]: I wonder if we should just add this CSS file instead of in HTML?

**Review by driskull — APPROVED** [review]
👍

File: packages/calcite-components/.storybook/utils.tsx (hunk 2 above)
Comment (jcfranco) [reply packages/calcite-components/.storybook/utils.tsx:193]: Do you mean as a stylesheet import in [`preview-html`](https://github.com/Esri/calcite-design-system/blob/main/packages/calcite-components/.storybook/preview-head.html)? FWIW, I went this route to keep the breakpoint-story styling closer to its usage, but I agree it'd be nice to have a single storybook CSS file to reference. I'll create a refactor issue to extract all styling to a global storybook stylesheet.

**Pull request merged**